recipes/ffmpeg/all/conanfile.py

```diff
@@ -339,41 +337,40 @@ def _target_arch(self):
+        if is_msvc(self):
             return "win32"
-        else:
-            triplet = tools.get_gnu_triplet(
-                "Macos" if is_apple_os(self) else str(self.settings.os),
-                str(self.settings.arch),
-                str(self.settings.compiler) if self.settings.os == "Windows" else None,
-            )
-            target_os = triplet.split("-")[2]
-            if target_os == "gnueabihf":
-                target_os = "gnu" # could also be "linux"
-            if target_os.startswith("android"):
-                target_os = "android"
-            return target_os
+        triplet = _get_gnu_triplet(
+            "Macos" if is_apple_os(self) else str(self.settings.os),
+            str(self.settings.arch),
+            str(self.settings.compiler) if self.settings.os == "Windows" else None,
+        )
+        target_os = triplet.split("-")[2]
+        if target_os == "gnueabihf":
+            target_os = "gnu" # could also be "linux"
+        if target_os.startswith("android"):
+            target_os = "android"
+        return target_os
 
     def _patch_sources(self):
-        if self._is_msvc and self.options.with_libx264 and not self.options["libx264"].shared:
+        if is_msvc(self) and self.options.with_libx264 and not self.options["libx264"].shared and Version(self.version) <= "5.0":
             # suppress MSVC linker warnings: https://trac.ffmpeg.org/ticket/7396
             # warning LNK4049: locally defined symbol x264_levels imported
             # warning LNK4049: locally defined symbol x264_bit_depth imported
-            tools.replace_in_file(os.path.join(self._source_subfolder, "libavcodec", "libx264.c"),
+            replace_in_file(self, os.path.join(self.source_folder, "libavcodec", "libx264.c"),
                                   "#define X264_API_IMPORTS 1", "")
         if self.options.with_ssl == "openssl":
             # https://trac.ffmpeg.org/ticket/5675
             openssl_libraries = " ".join(
-                ["-l%s" % lib for lib in self.deps_cpp_info["openssl"].libs])
-            tools.replace_in_file(os.path.join(self._source_subfolder, "configure"),
+                [f"-l{lib}" for lib in self.deps_cpp_info["openssl"].libs])
+            replace_in_file(self, os.path.join(self.source_folder, "configure"),
                                   "check_lib openssl openssl/ssl.h SSL_library_init -lssl -lcrypto -lws2_32 -lgdi32 ||",
-                                  "check_lib openssl openssl/ssl.h OPENSSL_init_ssl %s || " % openssl_libraries)
+                                  f"check_lib openssl openssl/ssl.h OPENSSL_init_ssl {openssl_libraries} || ")
 
     @contextlib.contextmanager
     def _build_context(self):
-        with tools.environment_append({"PKG_CONFIG_PATH": tools.unix_path(self.build_folder)}):
-            if self._is_msvc:
-                with tools.vcvars(self):
+        with environment_append({"PKG_CONFIG_PATH": unix_path(self, self.build_folder)}):
```

Comment:
```suggestion
        with environment_append({"PKG_CONFIG_PATH": unix_path(self, self.generators_folder)}):
```

Reply:
I don't think it's a good idea to use unix_path when recipe is not migrated to conan v2 build helpers.